addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-connection-service.js

```diff
@@ -220,6 +229,90 @@ class EnvironmentScConnectionService extends Service {
     return _.get(result, 'connection') || connection;
   }
 
+  async createConnectionSSMUrl(requestContext, envId, connectionId) {
+    const [environmentScService, pluginRegistryService] = await this.service([
+      'environmentScService',
+      'pluginRegistryService',
+    ]);
+
+    const connection = await this.mustFindConnection(requestContext, envId, connectionId);
+
+    // Write audit event
+    await this.audit(requestContext, { action: 'ssm-url-requested', body: { id: envId, connection } });
+
+    const { outputs } = await environmentScService.mustFind(requestContext, { id: envId });
+
+    const [aws] = await this.service(['aws']);
+
+    const { WorkspaceSSMRoleArn, Ec2WorkspaceInstanceId, SessionDuration } = cfnOutputsArrayToObject(outputs);
+
+    const { accessKeyId, secretAccessKey, sessionToken } = await aws.getCredentialsForRole({
+      roleArn: WorkspaceSSMRoleArn,
+    });
+
+    connection.url = await this.generateSigninSSMConsoleLink(
+      accessKeyId,
+      secretAccessKey,
+      sessionToken,
+      Ec2WorkspaceInstanceId,
+      SessionDuration,
+    );
+
+    // This is done so that plugins know it was called during create URL cycle
+    connection.operation = 'create';
+    // Give plugins chance to adjust the connection (such as connection url etc)
+    const result = await pluginRegistryService.visitPlugins(
+      'env-sc-connection-url',
+      'createConnectionUrl',
+      {
+        payload: {
+          envId,
+          connection,
+        },
+      },
+      { requestContext, container: this.container },
+    );
+
+    return _.get(result, 'connection') || connection;
+  }
+
+  async generateSigninSSMConsoleLink(accessKeyId, secretAccessKey, sessionToken, instanceId, inputSessionDuration) {
+    const session = {
+      sessionId: accessKeyId,
+      sessionKey: secretAccessKey,
+      sessionToken,
+    };
+
+    let sessionDuration = 3599; // default session duration, 59:59
+    if (Number(inputSessionDuration) > 0) {
```

Comment:
also should check the maximum allowed value of session duration

Reply:
I already limit maximum in linux Cloudformation template input parameter。

Reply:
The workspace template could be developed by third party, adding a sanity checking still makes sense.

Reply:
Sure, added it.